packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -263,26 +379,42 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
 
   // When the value prop changes, update the selectedIndex
   React.useEffect(() => {
-    dispatch([
-      'select',
-      options.findIndex((option) => option.value === valueProp),
-    ]);
-  }, [options, valueProp]);
-
-  // Call user-defined onChange when the value actually changes
-  React.useEffect(() => {
-    // Prevent user-defined onChange to be called on mount
-    if (!mounted.current) {
-      mounted.current = true;
-      return;
+    if (isMultipleEnabled(valueProp, multiple) && !!valueProp) {
+      const indexes = valueProp.map((value) => {
+        return options.findIndex((option) => option.value === value);
+      });
+      dispatch({ type: 'multi-override', value: indexes });
+    } else {
+      dispatch({
+        type: 'select',
+        value: options.findIndex((option) => option.value === valueProp),
+      });
     }
-    const currentValue = optionsRef.current[selectedIndex]?.value;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [valueProp, options]);
```

Comment:
why did you disable eslint here? putting `multiple` in this deps array makes sense to me

Reply:
Removed eslint line